pkg/operator/mon/health.go

```diff
@@ -80,6 +82,31 @@ func (c *Cluster) checkHealth() error {
 		} else {
 			logger.Warningf("mon %s NOT found in quorum. %+v", mon.Name, status)
 
+			var nodeState bool
+			if nodeState, err = c.getMonNodeStatus(mon.Name); err != nil {
+				logger.Errorf("failed to get mon %s node status. %+v", mon.Name, err)
+			}
+			// if the nodeState is false (= not ready) the mon is added to the
+			// list with the current time
+			if !nodeState {
+				if _, ok := c.monTimeoutList[mon.Name]; !ok {
+					c.monTimeoutList[mon.Name] = time.Now()
+				}
+
+				// when the timeout for the mon has been reached, continue to the
+				// normal failover/delete mon pod part of the code
+				if time.Since(c.monTimeoutList[mon.Name]) > monOutTimeout {
+					logger.Infof("mon %s node is not ready, in mon out timeout", mon.Name)
```

Comment:
I wonder if we should increase the time window for failover to all scenarios, instead of only the node not being in the `Ready` state. There will be many reasons why the mon falls out of quorum such as network blips, node failures, system hangs. We will never be able to predict them, so I'm not sure it's worth trying to find the special cases. What if we just use the timeout logic and not check the node state?

Reply:
Yes, using just the timeout logic here seems better, so we will not rely on having every case/state covered. I'll update the logic a bit and remove the `getMonNodeStatus()` function.